recipes-bsp/acpi-tables/samples/edison/leds.asl

```diff
@@ -0,0 +1,88 @@
+/*
+ * Intel Edison
+ *
+ * This table is modified from the Minnowboard Turbot leds.asl for Edison.
+ * 
+ * According to Intel® Edison Kit for Arduino Hardware Guide March 2017
+ * DS2 is the standard LED on the Arduino* board.
+ * It runs using the ‘blink’ code or whenever Digital I/O 13 is asserted High. 
+ * It can be used as an indicator under direct control.
+ * 
+ * This adds GPIO LEDs device for the DS2 LED and sets default heartbeat
+ * trigger for that.
+ *
+ * In Linux you need to set CONFIG_LEDS_GPIO=y (or m) to be able to use
+ * this device.
+ *
+ * Copyright (C) 2016, Intel Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+DefinitionBlock ("leds.aml", "SSDT", 5, "", "LEDS", 1)
```

Comment:
First of all, since it's an existing LED on Edison/Arduino we are referring to here, I would rather to split this to *leds.asl*:

    DefinitionBlock (...)
    {
      #include "arduino.asli"
      #include "ds2-led.asli"
    }

and move the rest to *ds2-led.asli*.

Otherwise you need to remove all mentions about Arduino and tell about GPIO pin numbering as per 70-pin Edison header.

Reply:
So how is that supposed to work then? Right now I just load arduino.aml followed by leds.aml which is nice and modular. And then I could unload leds and load spidev.

Maybe I don't understand yet how this is supposed to work?

ds2-led.asl would be a good name though, leds.asl would more fit a table for multiple leds

Reply:
The _*.asli_ files are the build blocks for "solution" _*.aml_. From the user prospective there is another layer, i.e. when we have _*.aml_ which can be chained (loaded sequentially). Thus we have to provide here (yes, seems above I was a bit wrong) leds.asli (note **i**), and provide ds2-led.asl (which would be considered as an example of the LED on Edison/Arduino) and, if you want to, leds.asl (note **no i**) as a generic example of the LED connected to the Edison by corresponding GPIO line.

Reply:
This should be OK, but I will test. However, while we are at it, there is another issue...